diff --git a/bin/internal/update_engine_version.sh b/bin/internal/update_engine_version.sh
index db05771c28..32cf32654a 100755
--- a/bin/internal/update_engine_version.sh
+++ b/bin/internal/update_engine_version.sh
@@ -16,9 +16,15 @@ set -e
 
 FLUTTER_ROOT="$(dirname "$(dirname "$(dirname "${BASH_SOURCE[0]}")")")"
 
-# Test for fusion repository
-if [ -f "$FLUTTER_ROOT/DEPS" ] && [ -f "$FLUTTER_ROOT/engine/src/.gn" ]; then
-  BRANCH=$(git -C "$FLUTTER_ROOT" rev-parse --abbrev-ref HEAD)
+# On stable, beta, and release tags, the engine.version is tracked by git - do not override it.
+TRACKED_ENGINE="$(git -C "$FLUTTER_ROOT" ls-files bin/internal/engine.version)"
+if [[ -n "$TRACKED_ENGINE" ]]; then
+  exit
+fi
+
+# Test for fusion repository and no environment variable override.
+if [ -z "$ENGINE_VERSION" ] && [ -f "$FLUTTER_ROOT/DEPS" ] && [ -f "$FLUTTER_ROOT/engine/src/.gn" ]; then
+
   # In a fusion repository; the engine.version comes from the git hashes.
   if [ -z "${LUCI_CONTEXT}" ]; then
     set +e
@@ -36,13 +42,11 @@ if [ -f "$FLUTTER_ROOT/DEPS" ] && [ -f "$FLUTTER_ROOT/engine/src/.gn" ]; then
     ENGINE_VERSION=$(git -C "$FLUTTER_ROOT" rev-parse HEAD)
   fi
 
-  if [[ "$BRANCH" != "stable" && "$BRANCH" != "beta" ]]; then
-    # Write the engine version out so downstream tools know what to look for.
-    echo $ENGINE_VERSION > "$FLUTTER_ROOT/bin/internal/engine.version"
+# Write the engine version out so downstream tools know what to look for.
+echo $ENGINE_VERSION > "$FLUTTER_ROOT/bin/internal/engine.version"
+
+# The realm on CI is passed in.
+if [ -n "${FLUTTER_REALM}" ]; then
+  echo $FLUTTER_REALM > "$FLUTTER_ROOT/bin/internal/engine.realm"
 
-    # The realm on CI is passed in.
-    if [ -n "${FLUTTER_REALM}" ]; then
-      echo $FLUTTER_REALM > "$FLUTTER_ROOT/bin/internal/engine.realm"
-    fi
-  fi
 fi
